### PR TITLE
BTRX-760: ORSP: Convert "My Task List" AND "My Projects" page

### PR DESCRIPTION
**Pull request by vvicario**
## Addresses
https://broadinstitute.atlassian.net/browse/BTRX-761
https://broadinstitute.atlassian.net/browse/BTRX-760
## Changes
Add a new component IssueList to show my projects and my task list. Remove gsp views that are not been used
## Testing
Describe for the reviewer how to test the specifics of this PR, both positive and negative cases.

---

- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Get a thumb from Belatrix
- [ ] **Submitter**: Get a thumb from @rushtong
- [ ] **Submitter**: Merge to develop using github's "Squash and Merge" feature
- [ ] **Submitter**: Delete branch after merge

### CODE REVIEW

File: src/main/webapp/issueList/IssueList.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
+import { Component } from 'react';
+import { hh, div, h1, h } from 'react-hyperscript-helpers';
+import { TableComponent } from "../components/TableComponent";
+import { Project } from '../util/ajax';
+import { Spinner } from "../components/Spinner";
+import { spinnerService } from "../util/spinner-service";
+import { formatDataPrintableFormat } from "../util/TableUtil";
+import { printData } from "../util/Utils";
+import { Link } from 'react-router-dom';
+import { format } from 'date-fns';
```
Comment (tadeork) [inline src/main/webapp/issueList/IssueList.js]: `format` is not used

File: src/main/webapp/issueList/IssueList.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+    hidden: true,
+    csvExport: false
+  }, {
+    dataField: 'projectKey',
+    text: 'Project',
+    sort: true,
+    headerStyle: (column, colIndex) => {
+      return { width: styles.project.projectKeyWidth };
+    },
+    formatter: (cell, row, rowIndex, colIndex) =>
```
Comment (tadeork) [inline src/main/webapp/issueList/IssueList.js:48]: This links can be moved to a function like this 
https://github.com/broadinstitute/orsp-pub/pull/252/files#diff-a9f4747606a7ea1e1584bf172822bf8fR50
```suggestion
        formatter: (cell, row, rowIndex, colIndex) => {
    return div({}, [
      linkFormatter(row, row.title)
    ])
  }
```

File: src/main/webapp/issueList/IssueList.js [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+    this._isMounted = false;
+  }
+
+  init = () => {
+    spinnerService.show(SPINNER_NAME);
+    this.tableHandler(0, this.state.sizePerPage, this.state.search, this.state.sort, this.state.currentPage);
+  };
+
+  tableHandler = (offset, limit, search, sort, page) => {
+    if(this._isMounted) {
```
Comment (tadeork) [inline src/main/webapp/issueList/IssueList.js]: `if (this._isMounted) { }` should be after the resolve of the promise is resolved

File: src/main/webapp/issueList/IssueList.js [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  init = () => {
+    spinnerService.show(SPINNER_NAME);
+    this.tableHandler(0, this.state.sizePerPage, this.state.search, this.state.sort, this.state.currentPage);
+  };
+
+  tableHandler = (offset, limit, search, sort, page) => {
```
Comment (tadeork) [inline src/main/webapp/issueList/IssueList.js]: This params seems to be unused. I think it's possible to remove it and call directly `tableHandler` from `componentDidMount`

File: src/main/webapp/issueList/IssueList.js [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+          sizePerPageList: SIZE_PER_PAGE_LIST,
+          pagination: true
+        }),
+        h(Spinner, {
+          name: SPINNER_NAME, group: "orsp", loadingImage: component.loadingImage
+        })
+      ])
+    )
+  }
+})
```
Comment (tadeork) [inline src/main/webapp/issueList/IssueList.js]: Missing semicolon.

File: src/main/webapp/rolesManagement/RolesManagement.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 import { Component } from 'react';
-import { h, div, span } from 'react-hyperscript-helpers';
+import { h, div, span, h1 } from 'react-hyperscript-helpers';
```
Comment (tadeork) [inline src/main/webapp/rolesManagement/RolesManagement.js]: `span` is unused.

**Review by tadeork — COMMENTED** [review]
One more comment about `RolesManagement` in the line 123 should be used `_isMounted` to prevent update unmounted component.

**Review by tadeork — APPROVED** [review]
:+1:

File: grails-app/controllers/org/broadinstitute/orsp/api/IssueListController.groovy [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import org.broadinstitute.orsp.utils.IssueUtils
+import org.broadinstitute.orsp.utils.UtilityClass
+
+@Slf4j
+@Resource
+class IssueListController extends AuthenticatedController {
+
+    def getProjectsForUser() {
+        try {
+            new UtilityClass(queryService).registerIssueListMarshaller()
```
Comment (rushtong) [inline grails-app/controllers/org/broadinstitute/orsp/api/IssueListController.groovy]: I missed this in https://github.com/broadinstitute/orsp-pub/pull/216. We should never be writing new constructors like this `new UtilityClass(queryService)`. Instead, `queryService` should be a spring-injected bean like we do in all of our other classes. If a class needs access to a bean, then it is no longer a utility style class and is now something that should be inside the grails world, i.e. a domain class, a service class, a controller class, a job class, etc.

File: grails-app/utils/org/broadinstitute/orsp/utils/UtilityClass.groovy [hunk 8]
```diff
@@ -150,6 +152,23 @@ class UtilityClass {
         }
     }
 
+    void registerIssueListMarshaller() {
+        JSON.createNamedConfig(ISSUE_LIST) {
+            SimpleDateFormat  sd = new SimpleDateFormat("yyyy-MM-dd")
+            it.registerObjectMarshaller( Issue ) { Issue ie ->
```
Comment (rushtong) [inline grails-app/utils/org/broadinstitute/orsp/utils/UtilityClass.groovy]: Standard convention is to just use `it ->`. In this case, `ie` doesn't seem to be an abbreviation for anything recognizable.

File: grails-app/utils/org/broadinstitute/orsp/utils/UtilityClass.groovy [hunk 9]
```diff
@@ -150,6 +152,23 @@ class UtilityClass {
...
+                return [
+                        id               : ie.id,
+                        projectKey       : ie.projectKey,
+                        summary          : IssueUtils.escapeQuote(ie.summary),
+                        status           : IssueUtils.escapeQuote(ie.status),
+                        type             : IssueUtils.escapeQuote(ie.type),
+                        updateDate       : ie.updateDate ? sd.format(ie.updateDate) : '',
+                        actors           : queryService.findUsersInUserNameList((List<String>)ie.getActorUsernames())?.collect { it.displayName }
```
Comment (rushtong) [inline grails-app/utils/org/broadinstitute/orsp/utils/UtilityClass.groovy]: Why the need for casting here `(List<String>)ie.getActorUsernames()` ? Issue `getActorUsernames` already returns a `Collection<String>`

File: src/main/webapp/issueList/IssueList.js [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+  }
+
+  init = () => {
+    spinnerService.show(SPINNER_NAME);
+    this.tableHandler();
+  };
+
+  tableHandler = () => {
+      Project.getProjectByUser(this.paramsContext.get('assignee'), this.paramsContext.get('max')).then(result => {
+        if(this._isMounted) {
```
Comment (rushtong) [inline src/main/webapp/issueList/IssueList.js:129]: ~You probably would not need to check mounted state if this api call were instead moved to `componentDidMount()`~

On second thought, that might not work, `tableHandler` is called pretty often.

**Comment by vvicario** [comment]
@rushtong Greg I worked on your feedback, can you please check it when you have some time?

**Review by rushtong — APPROVED** [review]
👍

**Pull request merged**